gateway/ln-gateway/src/lightning/cln.rs

```diff
@@ -3,104 +3,23 @@ use std::sync::Arc;
 use std::time::Duration;
 
 use async_trait::async_trait;
-use fedimint_core::encoding::{Decodable, Encodable};
 use fedimint_core::task::{sleep, TaskGroup};
 use fedimint_core::util::SafeUrl;
-use fedimint_core::Amount;
-use fedimint_ln_common::PrunedInvoice;
 use futures::stream::BoxStream;
-use serde::{Deserialize, Serialize};
-use thiserror::Error;
 use tonic::transport::{Channel, Endpoint};
 use tonic::Request;
 use tracing::info;
 
+use super::{ILnRpcClient, LightningRpcError};
 use crate::gateway_lnrpc::gateway_lightning_client::GatewayLightningClient;
 use crate::gateway_lnrpc::{
     EmptyRequest, EmptyResponse, GetNodeInfoResponse, GetRouteHintsRequest, GetRouteHintsResponse,
     InterceptHtlcRequest, InterceptHtlcResponse, PayInvoiceRequest, PayInvoiceResponse,
 };
-use crate::lnd::GatewayLndClient;
-use crate::LightningMode;
+use crate::lightning::MAX_LIGHTNING_RETRIES;
 pub type HtlcResult = std::result::Result<InterceptHtlcRequest, tonic::Status>;
 pub type RouteHtlcStream<'a> = BoxStream<'a, HtlcResult>;
```

Comment:
Should we move these to `gateway/ln-gateway-src-lightning/mod.rs`? They don't look CLN-specific.

Reply:
`cln.rs` is a misnomer, but these types relate to client implementation that uses grpc to connect to lightning service over a network hop.

This is not meant to be CLN specific

Reply:
Yeah that's a good point, it's a generic we use with cln but could be reapplied elsewhere, @okjodom what's a better naming scheme here?

Reply:
my alt name suggestion was `network.rs` but that has potential to be more confusing :)